parlai/core/torch_ranker_agent.py

```diff
@@ -31,9 +30,15 @@ def add_cmdline_args(argparser):
             choices=['batch', 'inline', 'fixed', 'vocab'],
             help='The source of candidates during training')
         agent.add_argument(
-            '-cands', '--fixed-candidates-path', type=str,
+            '-cands', '--fixed-candidates-path', type=str, default='',
             help='A text file of fixed candidates to use for all examples, one '
-                'candidate per line')
+                 'candidate per line')
+        agent.add_argument(
+            '-candvecs', '--fixed-candidate-vecs-path', type=str, default='',
```

Comment:
I'm tempted to suggest we just derive this path (use fixed_candidates_path or model_file with a special suffix), but that could be short sited. model_file + shorthash(fixed_candidate_paths)?

Reply:
Yeah, it does feel silly to have them type two probably very similar paths. But allowing a human-passed filepath lets them pass in candidates that they've encoded themselves offline if they'd like. Maybe we just assume the vec path is the same as the cand path with the .vec extension?

Reply:
I like the cache-and-read-from fixed-candidates-path.vec

Reply:
My concern for that was that fixed candidates is likely dataset specific (e.g. all the training examples) but the encodings will be model specific (every run in a sweep will have different caches!)

Reply:
How about this: -candvecs/--fixed-candidate-vectors is a string that's either 'reuse' [default], 'replace', or a path. 'reuse' will load from fixed-candidates-path.vec (or back off to 'replace' if that fails), 'replace' will generate a new vec file at fixed-candidates-path.vec, or if they pass a full path, we use whateever's there. So most of the time, if you're not getting fancy, you generated it once and then it always just reloads it. But if you really want to, you can keep separate vec files and switch between them by passing in their particular paths. Thoughts?

Reply:
I went ahead and implemented what I proposed above. You can take a look.